Userland/Services/LaunchServer/Launcher.cpp

```diff
@@ -227,14 +235,20 @@ Handler Launcher::get_handler_for_executable(Handler::Type handler_type, String
     return handler;
 }
 
-bool Launcher::open_with_user_preferences(HashMap<String, String> const& user_preferences, String const& key, Vector<String> const& arguments, String const& default_program)
+bool Launcher::open_with_user_preferences(HashMap<String, String>& user_preferences, String const& key, Vector<String> const& arguments, String const& default_program)
```

Comment:
Why must the user preferences not be const? Is `for_each_handler` modifying them; logically speaking it should not?

Reply:
That's right; I had made a change that would require it which I later removed, I fixed it back.